### PR TITLE
integrated my frontend and backend and implemented redux toolkit

### PR DESCRIPTION
**Pull request by Admayah**
Integrated my frontend app to my backend. So, user can create an account and login to their dashboard. They User(agent) can create, view, edit and delete property he/she posted on their dashboard.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,36 @@
...
-Runs the app in the development mode.\
-Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
+- [Reactjs](https://reactjs.org)
 
-The page will reload if you make edits.\
-You will also see any lint errors in the console.
+## Technology stack
 
-### `npm test`
+**9jaProperty** makes use of a host of modern technologies. The core ones are:
```
Comment (nosisky) [inline README.md:17]: Well done for updating the README.md with installation steps.

File: src/ProtectedRoute.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+// import { Navigate } from "react-router-dom";
+
+function ProtectedRoute({ children }) {
+    
+    if (!localStorage.getItem('token')) {
+        
+     window.location = '/'
+    }
+    console.log(localStorage.getItem('token'))
```
Comment (nosisky) [inline src/ProtectedRoute.js]: Take way the console.log statement if it's no longer needed.

File: src/components/allproperties/AllPropertiesData.jsx [hunk 3]
```diff
@@ -1,28 +1,48 @@
...
+import React, {useState, useEffect} from "react";
 import Navbar from "../Navbar/Navbar";
 import Footer from "../../components/footer/Footer";
 import AllProperties from "./allproperties";
 import "./propertiesdata.css";
 import PropertiesInfo from "./PropertiesInfo";
+import axios from 'axios'
+import { useSelector } from "react-redux";
+
+const url = 'http://localhost:4000/v1/properties'
```
Comment (nosisky) [inline src/components/allproperties/AllPropertiesData.jsx]: This API URL should be in an environment variable, research how to work with the environment variable in React. If the API URL changes, you don't always have to modify the code to reflect the changes.

File: src/components/editProperty/EditProperty.jsx [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+      ...editProperty, [e.target.name] : e.target.value
+    })
+  }
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+  
+
+      const response = await axios.put('http://localhost:4000/v1/agent/properties/' + id ,{...editProperty}, config)
```
Comment (nosisky) [inline src/components/editProperty/EditProperty.jsx]: Same thing here, the API URL should be in an environment variable.

File: src/components/editProperty/EditProperty.jsx [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+      dispatch(editStateProperty({
+        id,
+        ...editProperty
+      }))
+      navigate('/my-properties')
+      console.log(response.data)
+    } catch (error) {
+      console.log(error)
```
Comment (nosisky) [inline src/components/editProperty/EditProperty.jsx]: You are suppose to toast this error message to let the user know something Is wrong, your user won't be checking your console to figure out there is a problem.

**Review by nosisky — CHANGES_REQUESTED** [review]
Please request for more reviews from your teammates, there are so many things that can be improved about this code, check the code yourself and make necessary improvements.

File: src/components/allproperties/PropertiesInfo.jsx [hunk 6]
```diff
@@ -2,18 +2,19 @@ import React from "react";
 import { Link } from "react-router-dom";
 
 function PropertiesInfo(props) {
+  const {id, image, title, location, purpose} = props
   return (
     <>
-      <div className="property-info" id={props.id}>
+      <div className="property-info" id={id}>
         <div className="property-info-image">
           {" "}
```
Comment (olashehu) [inline src/components/allproperties/PropertiesInfo.jsx]: Remove this line

File: src/App.js [hunk 7]
```diff
@@ -13,30 +13,42 @@ import PropertyDetails from "./components/propertydetails/PropertyDetails";
...
+import 'react-toastify/dist/ReactToastify.css';
+import EditProperty from "./components/editProperty/EditProperty";
 
 function App() {
+  const notify = () => toast("Wow so easy!");
+
   return (
     <div className="App">
+    <button onClick={notify}>Notify!</button>
+    <ToastContainer />
```
Comment (olashehu) [inline src/App.js]: Please, remove this if you are done testing.

File: src/ErrorPage.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+export default function ErrorPage() {
+  return (
+    <div>There's nothing here: 404!</div>
```
Comment (olashehu) [inline src/ErrorPage.js]: I suggest we use something like "404 - Not Found"

File: src/components/allproperties/AllPropertiesData.jsx [hunk 9]
```diff
@@ -1,28 +1,48 @@
...
+
+  const getProperties = async () => {
+    const response =  await axios.get(url)
+    const {data} = response;
+    console.log('this is the array of all properties', data);
+    setProperties(data)
+  }
+
+  useEffect(() => {
+      // setProperties(AllProperties)
```
Comment (olashehu) [inline src/components/allproperties/AllPropertiesData.jsx]: Please, remove this if you are not using it.

File: src/components/allproperties/AllPropertiesData.jsx [hunk 10]
```diff
@@ -1,28 +1,48 @@
...
 import PropertiesInfo from "./PropertiesInfo";
+import axios from 'axios'
+import { useSelector } from "react-redux";
+
+const url = 'http://localhost:4000/v1/properties'
 
 function AllPropertiesData() {
+
+//  const {propertyItems} = useSelector((store) => store.properties)
+//  console.log(propertyItems)
```
Comment (olashehu) [inline src/components/allproperties/AllPropertiesData.jsx]: Please, remove this if you are no longer using it.

File: src/components/editProperty/EditProperty.jsx [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+                  placeholder="price"
+                  name="price"
+                  value={editProperty.price}
+                  onChange={editHandler}
+                />
+              </div>
+              <div className="property-data-input-field">
+                <select
+                  name="purpose"
+                  id="choces"
```
Comment (olashehu) [inline src/components/editProperty/EditProperty.jsx]: Typo

File: src/components/feed/Feed.jsx [hunk 12]
```diff
@@ -1,60 +1,138 @@
...
+
+
+const url = 'http://localhost:4000/v1/agent/properties';
+
+
 
 function Feed() {
+
+  const {id} = useParams()
+  console.log('----->', id)
```
Comment (olashehu) [inline src/components/feed/Feed.jsx]: Take way the console

File: src/components/feed/Feed.jsx [hunk 13]
```diff
@@ -1,60 +1,138 @@
...
+    }
+  }
+
+    const getProperties = async () => {
+      try {
+        const response =  await axios.get(url, config);
+        const {data} = await response;
+        setAgentProperties(data)
+        // dispatch({agentProperties})
+        console.log(data)
```
Comment (olashehu) [inline src/components/feed/Feed.jsx:35]: Same here and line 38

File: src/components/feed/Feed.jsx [hunk 14]
```diff
@@ -1,60 +1,138 @@
...
       <div className="my-properties-wrapper">
         <Sidebar />
         <div className="feed-container">
           <div class="feed-card-wrapper">
-            {Properties.map((item) => {
+            {agentProperties.map(({id, image, purpose, title, address, state, landrea, description,yearBuild, baths, rooms, store, garage, price}) => {
               return (
                 <div class="property-content">
                   <div className="property-image">
                     {" "}
```
Comment (olashehu) [inline src/components/feed/Feed.jsx]: Take this line out

File: src/components/feed/Feed.jsx [hunk 15]
```diff
@@ -1,60 +1,138 @@
...
+   })
+   console.log(newAgentProperties)
+  //  setAgentProperties(newAgentProperties)
+      try {
+        const response = await axios.delete('http://localhost:4000/v1/agent/properties/' + id, config);
+        console.log(response.data)
+         setAgentProperties(newAgentProperties)
+        dispatch(removeProperty(id))
+      } catch (error) {
+        console.log(error)
```
Comment (olashehu) [inline src/components/feed/Feed.jsx]: Make use of toast here

File: src/components/feed/properties.js [hunk 16]
```diff
@@ -1,110 +1,133 @@
...
+    description: 'Loveley',
+    rooms: "1 bed",
+    baths: "3 baths",
+    garage: "1 garage",
+    store: '1 store'
     },
-  },
 ];
 
 export default Properties;
```
Comment (olashehu) [inline src/components/feed/properties.js]: This file should not be inside the component folder

File: src/components/propertiesInput/PropertiesInputData.jsx [hunk 17]
```diff
@@ -1,11 +1,67 @@
...
+  
+const handleSubmit = async (e) => {
+  e.preventDefault()
+  try {
+    const response = await axios.post(url,{...propertiesInfo}, config)
+    const {id} = await response.data[0]
+    dispatch(addProperty({
+      id,
+      ...propertiesInfo
+    }))
```
Comment (olashehu) [inline src/components/propertiesInput/PropertiesInputData.jsx]: Please, make use of toast for success and error messages here too.

File: src/components/propertiesInput/PropertiesInputData.jsx [hunk 18]
```diff
@@ -89,25 +192,35 @@ export default function PropertiesInputData() {
...
+                      onChange={informationHandler}
+                      // placeholder="Street / Estate / Neighbourhood"
                     />
                   </div>
                 </div>
               </div>
               <div className="properties-content-wrapper">
                 <div className="property-location">
-                  <div>Price</div>
+                  {/* <div>Price</div> */}
```
Comment (olashehu) [inline src/components/propertiesInput/PropertiesInputData.jsx]: If you are not using this, remove it.

File: src/components/propertiesInput/PropertiesInputData.jsx [hunk 19]
```diff
@@ -126,12 +239,17 @@ export default function PropertiesInputData() {
               </div>
               <div className="properties-content-wrapper">
                 <div className="property-location">
-                  <div className="property-location-header">Price</div>
+                  {/* <div className="property-location-header">Price</div> */}
```
Comment (olashehu) [inline src/components/propertiesInput/PropertiesInputData.jsx]: Same here as above

File: src/components/propertiesInput/PropertiesInputData.jsx [hunk 20]
```diff
@@ -126,12 +239,17 @@ export default function PropertiesInputData() {
...
                   <div className="property-data-input-field">
                     <input
                       type="text"
                       className="property-data"
-                      placeholder="Price"
+                      // placeholder=""
```
Comment (olashehu) [inline src/components/propertiesInput/PropertiesInputData.jsx]: Please, look in your code for anywhere you have a code commented out and remove

File: src/components/propertydetails/PropertyDetails.jsx [hunk 21]
```diff
@@ -5,6 +5,7 @@ import "./propertydetails.css";
 
 function PropertyDetails() {
   const { id } = useParams();
+  console.log(id)
```
Comment (olashehu) [inline src/components/propertydetails/PropertyDetails.jsx]: Take this away

File: src/components/rent/Rent.jsx [hunk 22]
```diff
@@ -10,9 +10,9 @@ function Rent() {
...
       <Navbar />
       <div className="all-properties-container">
-        <div class="properties-card-wrapper">
-          {AllRentProperties.map((item) => (
-            <div class="property-info">
+        <div className="properties-card-wrapper">
+          {AllRentProperties.map((item, index) => (
+            <div className="property-info" key={index}>
               <div className="property-info-image">
                 {" "}
```
Comment (olashehu) [inline src/components/rent/Rent.jsx]: Same here

File: src/components/sales/Sales.jsx [hunk 23]
```diff
@@ -10,9 +10,9 @@ function Sales() {
...
       <Navbar />
       <div className="all-properties-container">
-        <div class="properties-card-wrapper">
-          {AllSaleProperties.map((item) => (
-            <div class="property-info">
+        <div className="properties-card-wrapper">
+          {AllSaleProperties.map((item, index) => (
+            <div className="property-info" key={index}>
               <div className="property-info-image">
                 {" "}
```
Comment (olashehu) [inline src/components/sales/Sales.jsx]: Same here

**Review by olashehu — CHANGES_REQUESTED** [review]
Well done, I drop some comments.

File: src/components/Navbar/navbar.css [hunk 24]
```diff
@@ -1,9 +1,15 @@
...
   color: white;
   max-width: 1536px;
   margin: 0 auto;
+  border-bottom: 1px solid #F98547;
+  /* background-color: #F98547; */
+
+  /* position: fixed;
+  z-index: 2; */
+  /* top: 0; */
 }
```
Comment (mykelmatics) [inline src/components/Navbar/navbar.css:8]: remove commented lines

File: src/components/allproperties/AllPropertiesData.jsx [hunk 25]
```diff
@@ -1,32 +1,95 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import Navbar from "../Navbar/Navbar";
 import Footer from "../../components/footer/Footer";
-import AllProperties from "./allproperties";
-import "./propertiesdata.css";
 import PropertiesInfo from "./PropertiesInfo";
+import axios from 'axios'
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
```
Comment (mykelmatics) [inline src/components/allproperties/AllPropertiesData.jsx:7]: I suggest you have import of packages before component

File: src/components/allproperties/AllPropertiesData.jsx [hunk 26]
```diff
@@ -1,32 +1,95 @@
...
+  // const getProperties = async () => {
+  //   try {
+  //     const response = await axios.get()
+  //     const { data } = response;
+  //     setProperties(paginate(data))
+  //     toast('Property is successfully created')
+  //   } catch (error) {
+  //     toast.error('Something went wrong')
+  //   }
+  // };
```
Comment (mykelmatics) [inline src/components/allproperties/AllPropertiesData.jsx]: remove comment

File: src/components/allproperties/AllPropertiesData.jsx [hunk 27]
```diff
@@ -1,32 +1,95 @@
...
+      }
+      return prevPage
+    })
+  }
+
+  const handlePage = (index) => {
+    setPage(index)
+  }
+
+console.log(properties)
```
Comment (mykelmatics) [inline src/components/allproperties/AllPropertiesData.jsx:51]: I suggest you remove this

File: src/components/allproperties/propertiesdata.css [hunk 28]
```diff
@@ -37,6 +37,7 @@
 .property-img {
   width: 100%;
   height: 100%;
+  /* object-fit: cover; */
```
Comment (mykelmatics) [inline src/components/allproperties/propertiesdata.css:40]: remove comment

File: src/components/editProperty/EditProperty.jsx [hunk 29]
```diff
@@ -0,0 +1,270 @@
...
+
+  const {id} = useParams();
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  
+  const token = localStorage.getItem('token')
+    let config = {
+      "headers": {
```
Comment (mykelmatics) [inline src/components/editProperty/EditProperty.jsx:24]: Try to format your code

File: src/components/feed/Feed.jsx [hunk 30]
```diff
@@ -1,60 +1,135 @@
...
+import { removeProperty } from "../../features/properties/adminProperties";
+import { useDispatch } from "react-redux";
+import axios from 'axios'
+import { useNavigate } from "react-router-dom";
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+
+// const url = 'http://localhost:4000/v1/agent/properties';
+
```
Comment (mykelmatics) [inline src/components/feed/Feed.jsx:12]: remove comment

File: src/components/propertiesInput/PropertiesInputData.jsx [hunk 31]
```diff
@@ -1,170 +1,251 @@
...
+                onChange={informationHandler}
+              />
+              <label htmlFor="price">Title</label>
+              <InputField
+                type='text'
+                placeholder='price'
+                name='price'
+                value={propertiesInfo.price}
+                onChange={informationHandler}
+              />
```
Comment (mykelmatics) [inline src/components/propertiesInput/PropertiesInputData.jsx:114]: I suggest you map the Input so you can reduce the lines of code here and also stick to DRY principle

File: src/components/propertiesInput/PropertiesInputData.jsx [hunk 32]
```diff
@@ -1,170 +1,251 @@
...
+                />
+              <label htmlFor="price">land Area</label>
+               <InputField
+                type='text'
+                placeholder='land area'
+                name='land_area'
+                value={propertiesInfo.landArea}
+                onChange={informationHandler}
+                />
+              <label htmlFor="price">Property Description</label>
```
Comment (mykelmatics) [inline src/components/propertiesInput/PropertiesInputData.jsx:220]: I suggest you map the input field so you can reduce the lines of code here and also stick to DRY principle

File: src/components/propertydetails/Features.js [hunk 33]
```diff
@@ -1,23 +1,51 @@
...
 
 function Features(props) {
+
+  const { id } = useParams();
+  const [moreInfo, setMoreInfo] = useState([]);
+
+  const getProperty = async () => {
+    try {
+      const response = await axios.get(`${process.env.REACT_APP_BASEURL}/properties`)
+      console.log(response)
```
Comment (mykelmatics) [inline src/components/propertydetails/Features.js:15]: you can remove this

File: src/components/propertydetails/PropertyDetails.jsx [hunk 34]
```diff
@@ -1,18 +1,56 @@
...
+  // const {adminProperties} = useSelector((state) => state.adminProperties)
+// return item.filter((item) => item.id === id))
+// const singlePropertyInfo = moreInfo.((item )=> {
+//  if (item.id === parseInt(id)) {
+//   return 'Yes'
+//  } 
+//  return 'No'
+//   })
+// console.log(singlePropertyInfo)
+
```
Comment (mykelmatics) [inline src/components/propertydetails/PropertyDetails.jsx]: remove commented lines

File: src/components/rent/Rent.jsx [hunk 35]
```diff
@@ -1,20 +1,119 @@
...
+  //     toast.error('Something went wrong') 
+  //   }
+  // };
+
+  // useEffect(() => {
+  //   getRentProperties()
+  // }, []);
+
+  // console.log(rentProps)
+
```
Comment (mykelmatics) [inline src/components/rent/Rent.jsx]: same here

File: src/components/sales/Sales.jsx [hunk 36]
```diff
@@ -1,20 +1,117 @@
...
+  //   } catch (error) {
+  //     toast.error('Something went wrong') 
+  //   }
+  // };
+
+  // useEffect(() => {
+  //   getRentProperties()
+  // }, []);
+
+  // console.log(rentProps)
```
Comment (mykelmatics) [inline src/components/sales/Sales.jsx]: same here

File: src/pages/profile/Profile.jsx [hunk 37]
```diff
@@ -52,45 +77,42 @@ function Profile() {
                       id=""
                     />
                   </span>
-                </div>
-                <div className="property-infos">
+                </div> */}
+                {/* <div className="property-infos">
```
Comment (mykelmatics) [inline src/pages/profile/Profile.jsx]: remove commented line

File: src/store.js [hunk 38]
```diff
@@ -0,0 +1,13 @@
+import { configureStore } from '@reduxjs/toolkit';
+import propertyReducer from './features/properties/propertiesSlice'
+import adminProperties from './features/properties/adminProperties';
+import userReducer from './features/properties/userSlice';
+// import userReducer from './features/auth/slice';
```
Comment (mykelmatics) [inline src/store.js:5]: same here

**Review by mykelmatics — COMMENTED** [review]
Overall this looks good, The style and all.

**Pull request merged**